core/dbt/context/common.py

```diff
@@ -447,6 +435,10 @@ def generate_model(model, config, manifest, source_config, provider):
     model_dict = model.to_dict()
     context = generate_base(model, model_dict, config, manifest,
                             source_config, provider)
+    # operations (hooks) don't get a 'this'
+    if model.resource_type != NodeType.Operation:
+        this = get_this_relation(context['adapter'], config, model_dict)
+        context['this'] = this
```

Comment:
If someone had `{{this}}` in a hook before it would spit out a dummy relation, right? While this is correct, it's probably (stupidly) a breaking change that we should note in the changelog... can you do that before you merge this so that we don't forget?